source/modulo_components/modulo_components/component.py

```diff
@@ -49,13 +49,18 @@ def execute(self):
         self.__execute_thread = Thread(target=self.__on_execute)
         self.__execute_thread.start()
 
-    def __on_execute(self):
+    def __on_execute(self) -> None:
         """
         Run the execution function in a try catch block and set the predicates according to the outcome of the
         execution.
         """
         try:
-            if not self.on_execute_callback():
+            result = self.on_execute_callback()
+            if result is None:
+                self.get_logger().error("'on_execute_callback' doesn't return a value, error will be raised.")
```

Comment:
Some wording suggestions - I would prefer it like this:
```suggestion
                self.get_logger().error("Expected a return value from 'on_execute_callback'")
```

The current error text alone ("callback doesn't return a value") could potentially be ambiguous (it didn't return a value but it should, vs. it returned a value but it shouldn't).

And instead of including `error will be raised` everywhere, does `raise_error` have its own log entry? It would be nice not to repeat it.

Reply:
It has its log entry correct :+1: